### ISSUE TITLE
No request executed for update_billing_agreement method

### ISSUE THREAD

**Issue by GeekPress** [woocommerce/woocommerce-gateway-paypal-express-checkout#602]
Into the `WC_Gateway_PPEC_Client` class, the method `update_billing_agreement` doesn't execute the request.

This part is missing:
`return $this->_request( $params );`

**Comment by james-allan** [comment]
As far as I can tell, the `update_billing_agreement()` function is never used in the PPEC plugin code. Looking at the [PayPal documentation](https://developer.paypal.com/docs/classic/api/merchant/BAUpdate-API-Operation-NVP/) it can be used to: 

1. Update a billing agreement's status (to cancel it),
2. Update the billing agreement's description (`BILLINGAGREEMENTDESCRIPTION`),
3. or set a custom annotation on the agreement (`BILLINGAGREEMENTCUSTOM`). 

None of those args are used in `update_billing_agreement()` and there's no way a third-party could add them either so I'm pretty sure this function wouldn't do much in the way of updating anyway, even if we did fire off the request. I suspect it was a placeholder that was never implemented.

So our options are to:

1. Fire off the request by adding `return $this->_request( $params );`. PayPal will respond with general information about the billing agreement but no update will be made. See example response below.
2. Deprecate the function since it not used and cannot really be used in a way that makes sense for the function name and documented description. 

```
WC_Gateway_PPEC_Client::_process_response: acknowledged response body: Array
(
    [BILLINGAGREEMENTID] => B-77W32XXXXXXXXXX
    [BILLINGAGREEMENTDESCRIPTION] => Orders with wc-dev
    [BILLINGAGREEMENTSTATUS] => Active
    [BILLINGAGREEMENTCUSTOM] => {"order_id":123,"order_key":"wc_order_1234"}
    [TIMESTAMP] => 2020-01-30T02:09:43Z
    [CORRELATIONID] => XXXXXX
    [ACK] => Success
    [VERSION] => 120.0
    [BUILD] => 54085969
    [EMAIL] => XXXX
    [PAYERID] => XXXX
    [PAYERSTATUS] => verified
    [FIRSTNAME] => James
    [LASTNAME] => Allan
    [COUNTRYCODE] => GB
    [ADDRESSID] => PayPal
)
```

**Comment by mattallan** [comment]
Closed with #673
